### DISCUSSION TITLE
What's the use of RoslynString class

### DISCUSSION THREAD

**Discussion by Youssef1313** [dotnet/roslyn#47206] [General] [answered]
https://github.com/dotnet/roslyn/blob/e704ca635bd6de70a0250e34c4567c7a28fa9f6d/src/Compilers/Core/Portable/InternalUtilities/RoslynString.cs#L1-L23

Why not the consumers just use string.IsNullOrEmpty and string.IsNullOrWhiteSpace?

**Comment by CyrusNajmabadi** [comment, marked as answer]
It provides nullable annotations for when we weren't targetting reference libraries with nullable annotations.

**Reply by Youssef1313** [reply]
@CyrusNajmabadi Thanks!
Is it still required given that runtime have them annotated now?

**Reply by jmarolf** [reply]
I do not believe the runtime has annotated netstandard2.0 or any of the .NET Framework targets
